### PR TITLE
Update webhook POST to send application/json Content-Type

### LINKED ISSUE #5814: Order creation webhook doesn't send data with "application/json"
### What I'm trying to achieve
Send order data to a web server via webhooks

### Steps to reproduce the problem
1. Create a simple server that prints out the JSON request body
2. Create a order creation webhook pointing to this server
3. Create a new order
4. Observe that the JSON request body sent to the server is empty

### What I expected to happen
Order data should be sent as JSON body

### Screenshots
image

Second line is `console.log(req.body)` in an example expressjs server with `body-parser` JSON parser

After modifying the file in `saleor/plugins/webhook/tasks.py` to add the `Content-Type: application/json` header when sending the data, it worked as expected.

### PR DESCRIPTION
**Pull request by koradon**
I want to merge this change because...
Using `json` instead of `data` will automatically do what we want.
Link to requests [docs](https://requests.kennethreitz.org/en/master/user/quickstart/#more-complicated-post-requests)

fixes #5814

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by koradon** [comment]
@NyanKiyoshi It is `mock_request.assert_called_once_with`. Rest of validation is made by VCR which I need to overwrite

File: saleor/plugins/webhook/tests/test_webhook.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def test_trigger_webhooks_for_event(
     }
 
     mock_request.assert_called_once_with(
-        webhook.target_url, data=expected_data, headers=expected_headers, timeout=10
+        webhook.target_url, json=expected_data, headers=expected_headers, timeout=10
```
Comment (korycins) [inline saleor/plugins/webhook/tests/test_webhook.py:44]: Maybe I don't remember but I don't understand why we are mocking a request if we have enabled `vcr`

Comment (koradon) [reply saleor/plugins/webhook/tests/test_webhook.py:44]: Well if I have to guess it is just another check for arguments for request POST

Comment (NyanKiyoshi) [reply saleor/plugins/webhook/tests/test_webhook.py:44]: Shouldn't the tests fail if the VCR are outdated?

Comment (koradon) [reply saleor/plugins/webhook/tests/test_webhook.py:44]: @NyanKiyoshi yes and it fails as you can see above.

Comment (korycins) [reply saleor/plugins/webhook/tests/test_webhook.py:44]: Ach, yes! I've just saw the `wraps`

Comment (koradon) [reply saleor/plugins/webhook/tests/test_webhook.py:44]: @NyanKiyoshi wait what... on test passed while on local it was failing. I have just recreated VCRs...
The second commit should fail.

**Comment by koradon** [comment]
Coverage is failing because I edited `tox.ini`

**Pull request merged**